src/Block/Fetch.hs

```diff
@@ -60,170 +65,272 @@ import Block.Types (
   mkFindIntersectRequest,
   mkRequestNextRequest,
  )
-import Database (saveDatums, updateLastBlock)
+import Database (getLastBlock, saveDatums, updateLastBlock)
 
 data OgmiosInfo = OgmiosInfo
   { ogmiosPort :: Int
   , ogmiosAddress :: String
   }
 
-newtype OgmiosWorkerMVar = MkOgmiosWorkerMVar (MVar (Async ()))
-
-data StartBlockFetcherError
-  = StartBlockFetcherErrorAlreadyRunning
-  deriving stock (Show)
+data BlockFetcherEnv = BlockFetcherEnv
+  { -- | Queue used to push blocks for block processor.
+    queue :: TBQueue AlonzoBlock
+  , -- | WS connection to ogmios. Should be used only via 'sendAndReceive'.
+    wsConnMVar :: MVar WebSockets.Connection
+  , -- | Intersection from which block fetcher will start fetching blocks.
+    -- Should be initialized only via `findIntersection`.
+    intersectionTVar :: TVar (Maybe CursorPoint)
```

Comment:
Maybe it will be better to rename this field to ogmiosCursor or something like this, here and in the related places. That creates a gap between ODC and ogmios terminology but should simplify code, but such terminology inside makes ODC simple.

Reply:
ogmiosCursor also isn't perfect because it's not updated as cursor changes when requesting next blocks. Naming is difficult